### PR TITLE
Migration plugin docs to integration framework

### PR DESCRIPTION
**Pull request by nywilken**
Add scaffolding for [HashiCorp integrations](https://github.com/hashicorp/integrations)

### CODE REVIEW

**Review by lbajolet-hashicorp — APPROVED** [review]
LGTM!

File: GNUmakefile [hunk 1]
```diff
@@ -35,3 +35,8 @@ generate: install-packer-sdc
 	packer-sdc renderdocs -src ./docs -dst ./.docs -partials ./docs-partials
 	# checkout the .docs folder for a preview of the docs
 
+build-docs: install-packer-sdc
```
Comment (JenGoldstrich) [inline GNUmakefile:38]: Why do we still leave the older generate task in place, should we only use build-docs after this PR is merged?

**Review by BrandonRomano — APPROVED** [review]

File: GNUmakefile (hunk 1 above)
Comment (nywilken) [reply GNUmakefile:38]: If you're referring to the release pipeline. We are keeping everything the same until we've cut over the plugins to production. Once live we can remove the doc.zip generation. But it should not affect anything if left in place.

File: docs/README.md [hunk 2]
```diff
@@ -1,69 +1,89 @@
...
-Then, run [`packer init`](https://www.packer.io/docs/commands/init).
+To install this plugin, copy and paste this code into your Packer configuration, then run [`packer init`](https://www.packer.io/docs/commands/init).
 
 ```hcl
 packer {
   required_plugins {
     azure = {
-      version = ">= 1.4.2"
       source  = "github.com/hashicorp/azure"
+      version = "~> 1"
```
Comment (lbajolet-hashicorp) [inline docs/README.md:13]: Now that v2 is out, we could update this block to point to this version I would think

Comment (nywilken) [reply docs/README.md]: Updated.

**Pull request merged**